### PR TITLE
made various fixes to make the percentage function work

### PR DESCRIPTION
**Pull request by HenokAddis**
- updated the function to call from estimate summary it works with no changes to the kpi python
- fixed test case to compare the pre-defined test case kpis called from the function with the logic we believe to be correct
- small fixes here and tehre

### CODE REVIEW

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py [hunk 1]
```diff
@@ -261,27 +265,30 @@ def printTreatmentMatrixPoints(self, df):
...
+                '}')
 
     def calculatePercentImprovement(self, df):
         '''
         Calculates and prints out the percent improvement of the treated group in comparison to comparison group.
         '''
         treatmentMatrix = self._calculate_aggregate_treatment_matrix(df, aggregate=np.mean)
         kpiTreat0Post0_mean, kpiTreat0Post1_mean, kpiTreat1Post0_mean, kpiTreat1Post1_mean = treatmentMatrix
-        inital = kpiTreat1Post0_mean + (kpiTreat0Post1_mean - kpiTreat0Post0_mean)
+        initial = kpiTreat1Post0_mean + (kpiTreat0Post1_mean - kpiTreat0Post0_mean)
```
Comment (csbuja) [inline LongitudinalDifferenceInDifferencesPy/didDesigner.py:282]: @jlbuja

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py [hunk 2]
```diff
@@ -176,6 +176,10 @@ def estimateSummmary(self,df):
         print("Area under the Curve Estimate:")
         print(self.AreaUnderCurveEstimate(df))
 
+        print("")
```
Comment (csbuja) [inline LongitudinalDifferenceInDifferencesPy/didDesigner.py]: Nice. I like this better than running the function individually. I have to edit the wiki to reflect this change.

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py [hunk 3]
```diff
@@ -378,13 +385,11 @@ def printPercentImprovement(self, df):
...
-            '\npost-treatment mean KPI of treated group: ' + str(kpiTreat1Post1_mean))
+    
+    #this needs to be updated to call the function
+    kpiTreat0Post0_mean, kpiTreat0Post1_mean, kpiTreat1Post0_mean, kpiTreat1Post1_mean = 4.0, 9.0, 3.0, 13.0
     percentImprovement = ((kpiTreat1Post1_mean - (kpiTreat1Post0_mean + \
         (kpiTreat0Post1_mean - kpiTreat0Post0_mean))) / (kpiTreat1Post0_mean + \
             (kpiTreat0Post1_mean - kpiTreat0Post0_mean)))*100
-    print('percent improvement: ' + str(percentImprovement) + '%')
-    assert(percentImprovement == ((2 / 3) * 100))
+    assert(percentImprovement==obj.calculatePercentImprovement(df))
```
Comment (csbuja) [inline LongitudinalDifferenceInDifferencesPy/didDesigner.py]: Thanks for calling the function in the test case. I'm not sure why this didn't happen before. 

I will say that that this assert statement isn't quite doing the right thing. We should be asserting that `((2 / 3) * 100) == obj.calculatePercentImprovement(df))`

Comment (csbuja) [reply LongitudinalDifferenceInDifferencesPy/didDesigner.py]: Please add spaces before and after == operator

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py [hunk 4]
```diff
@@ -261,27 +265,30 @@ def printTreatmentMatrixPoints(self, df):
...
         return percentImprovement
 
     def printPercentImprovement(self, df):
         '''
         Prints percent improvent of the treated group in comparison to comparison group.
         '''
-        percentImprovement = calculatePercentImprovement(self, df)
-        print('percent improvement: ' + str(percentImprovement) + '%')
+        percentImprovement = self.calculatePercentImprovement(df)
+        self.printTreatmentMatrixPoints(df)
```
Comment (csbuja) [inline LongitudinalDifferenceInDifferencesPy/didDesigner.py]: please change function name if we want to keep this line

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py [hunk 5]
```diff
@@ -261,27 +265,30 @@ def printTreatmentMatrixPoints(self, df):
         treatmentMatrix = self._calculate_aggregate_treatment_matrix(df, aggregate=np.mean)
         kpiTreat0Post0_mean, kpiTreat0Post1_mean, kpiTreat1Post0_mean, kpiTreat1Post1_mean = \
             treatmentMatrix
-        print('pre-treatment mean KPI of comparison group: ' + kpiTreat0Post0_mean + \
-            '\npost-treatment mean KPI of comparison group: ' + kpiTreat0Post1_mean + \
-            '\npre-treatment mean KPI of treated group: ' + kpiTreat1Post0_mean + \
-            '\npost-treatment mean KPI of treated group: ' + kpikpiTreat1Post1_mean)
+
+        print('{ Pre-treatment mean KPI of comparison group: ' + str(kpiTreat0Post0_mean) + \
```
Comment (csbuja) [inline LongitudinalDifferenceInDifferencesPy/didDesigner.py:270]: This looks better, for sure.

**Review by csbuja — CHANGES_REQUESTED** [review]
- Please fix the assert in the test case. 
- General feedback, add spaces before and after operators.

- change function name of printPercentImprovement since you're not just printing the percent improvement.... you're also printing the treatment matrix.

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py (hunk 2 above)
Comment (HenokAddis) [reply LongitudinalDifferenceInDifferencesPy/didDesigner.py]: I made a minor edit here, but same behavior

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py (hunk 3 above)
Comment (HenokAddis) [reply LongitudinalDifferenceInDifferencesPy/didDesigner.py]: Why is the correct assertion (2/3)*100? That isn't the correct output given the formula.

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py (hunk 4 above)
Comment (HenokAddis) [reply LongitudinalDifferenceInDifferencesPy/didDesigner.py]: I separated the output to be print treatment matrix and print percent improvement

File: LongitudinalDifferenceInDifferencesPy/didDesigner.py (hunk 3 above)
Comment (HenokAddis) [reply LongitudinalDifferenceInDifferencesPy/didDesigner.py]: I updated to (5/8)*100 which is the correct result

**Comment by HenokAddis** [comment]
I have applied the fixes you recommended. Please give one final review before closing the PR

**Pull request merged**